blivet/devicelibs/disk.py

```diff
@@ -0,0 +1,110 @@
+from collections import namedtuple
+
+from . import raid
+from .. import errors
+from .. import util
+from ..size import Size
+
+lsm = None
+
+_HBA_PLUGIN_URIS = ("hpsa://", "megaraid://")
+LSMInfo = namedtuple('HBAVolumeInfo', ['system', 'nodes', 'raid_type', 'raid_stripe_size', 'raid_disk_count'])
+""" .. class:: LSMInfo
+
+        .. attribute:: system (str): descriptive name of HBA unit
+        .. attribute:: nodes (list[str]): list of device node paths for the volume
+        .. attribute:: raid_type (:class:`~.devicelibs.raid.RAIDLevel` or None): RAID level
+        .. attribute:: raid_stripe_size (:class:`~.size.Size` or None): stripe size
+        .. attribute:: raid_disk_count (int or None): number of disks in the RAID set
+"""
+volumes = dict()
+_raid_levels = dict()
+
+
+class _LSMRAIDLevelStub(raid.RAIDLevel):
+    def __init__(self, name):
+        self._name = name
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def names(self):
+        return [self.name]
+
+    @property
+    def min_members(self):
+        return 0
+
+    def has_redundancy(self):
+        return False
+
+    def is_uniform(self):
+        return False
+
+
+class _LSMDependencyGuard(util.DependencyGuard):
+    error_msg = "libstoragemgmt functionality not available"
+
+    def _check_avail(self):
+        global lsm
+        try:
+            import lsm  # pylint: disable=redefined-outer-name
```

Comment:
It is provided by `libstoragemgmt-python`. I am trying to avoid adding any hard dependencies, and made an effort to arrange things to function without failure whether or not libstoragemgmt is installed (or its service running). I am open to argument, of course, if you think it should be a hard requirement for all blivet users.